src/query_plan/operation.rs

```diff
@@ -517,6 +517,12 @@ pub(crate) trait HasNormalizedSelectionKey {
     fn key(&self) -> NormalizedSelectionKey;
 }
 
+impl HasNormalizedSelectionKey for NormalizedSelectionKey {
+    fn key(&self) -> NormalizedSelectionKey {
+        self.clone()
+    }
+}
```

Comment:
This looks a bit smelly. Why is it necessary?

Reply:
Probably should be removed. At one point, I thought I was going to need to use this, but I ended up not needing it.